main.go

```diff
@@ -20,68 +20,89 @@ import (
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
-	routev1 "github.com/openshift/api/route/v1"
-	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
-	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-
-	meteorv1alpha1 "github.com/aicoe/meteor-operator/api/v1alpha1"
-	common "github.com/aicoe/meteor-operator/controllers/common"
-	meteor "github.com/aicoe/meteor-operator/controllers/meteor"
-	shower "github.com/aicoe/meteor-operator/controllers/shower"
-	//+kubebuilder:scaffold:imports
+	meteorv1alpha1 "github.com/thoth-station/meteor-operator/api/v1alpha1"
+	"github.com/thoth-station/meteor-operator/controllers"
+	common "github.com/thoth-station/meteor-operator/controllers/common"
+	meteor "github.com/thoth-station/meteor-operator/controllers/meteor"
+	shower "github.com/thoth-station/meteor-operator/controllers/shower"
 )
 
 var (
-	scheme   = runtime.NewScheme()
-	setupLog = ctrl.Log.WithName("setup")
+	scheme     = runtime.NewScheme()
+	setupLog   = ctrl.Log.WithName("setup")
+	ctrlConfig = meteorv1alpha1.MeteorConfig{}
 )
 
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 	utilruntime.Must(meteorv1alpha1.AddToScheme(scheme))
 	utilruntime.Must(pipelinev1beta1.AddToScheme(scheme))
-	utilruntime.Must(routev1.AddToScheme(scheme))
+	if ctrlConfig.Spec.EnableShower {
+		utilruntime.Must(routev1.AddToScheme(scheme))
+	}
 	utilruntime.Must(monitoringv1.AddToScheme(scheme))
 	//+kubebuilder:scaffold:scheme
 	common.InitMetrics()
 }
 
 func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	var probeAddr string
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
+	var configFile string
+
+	flag.StringVar(&configFile, "config", "",
+		"The controller will load its initial configuration from this file. "+
+			"Omit this flag to use the default configuration values. "+
+			"Command-line flags override configuration from this file.")
 	opts := zap.Options{
 		Development: true,
 	}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	var err error
+
+	options := ctrl.Options{
+		Scheme:                  scheme,
+		MetricsBindAddress:      "127.0.0.1:8080",
+		Port:                    9443,
+		HealthProbeBindAddress:  ":8081",
+		LeaderElection:          true,
+		LeaderElectionID:        "05b1bff9.meteor.zone",
+		LeaderElectionNamespace: "cnbi-system",
```

Comment:
This has changed in `config/default/kustomization.yaml` and therefore needs a matching change here:

```suggestion
		LeaderElectionNamespace: "aicoe-meteor",
```

We should also find a way to have a single place to configure that... but meanwhile, let's manually keep in sync